model/annotations/task_annotations.go

```diff
@@ -204,6 +204,48 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 	return errors.Wrapf(err, "problem adding task annotation for '%s'", a.TaskId)
 }
 
+func AddToAnnotation(a *TaskAnnotation, userDisplayName string) error {
+	existingAnnotation, err := FindOneByTaskIdAndExecution(a.TaskId, a.TaskExecution)
+	if err != nil {
+		return errors.Wrapf(err, "finding task annotation for task id '%s'", a.TaskId)
+	}
+	if existingAnnotation == nil {
+		return errors.Wrapf(err, "missing task annotation for task id '%s'", a.TaskId)
+	}
+	source := &Source{
+		Author:    userDisplayName,
+		Time:      time.Now(),
+		Requester: APIRequester,
+	}
+	update := bson.M{
+		TaskIdKey:        a.TaskId,
+		TaskExecutionKey: a.TaskExecution,
+	}
+
+	if a.Issues != nil {
+		for i := range a.Issues {
+			a.Issues[i].Source = source
+			existingAnnotation.Issues = append(existingAnnotation.Issues, a.Issues[i])
+		}
+	}
+	update[IssuesKey] = existingAnnotation.Issues
```

Comment:
This should be in the if statement at 225, i.e. we should only overwrite issues if the user intended to.